pkg/services/ngalert/api/compat.go

```diff
@@ -0,0 +1,91 @@
+package api
+
+import (
+	"time"
+
+	"github.com/prometheus/common/model"
+
+	"github.com/grafana/grafana/pkg/services/ngalert/api/tooling/definitions"
+	"github.com/grafana/grafana/pkg/services/ngalert/models"
+)
+
+// AlertRuleFromProvisionedAlertRule converts definitions.ProvisionedAlertRule to models.AlertRule
+func AlertRuleFromProvisionedAlertRule(a definitions.ProvisionedAlertRule) (models.AlertRule, error) {
+	return models.AlertRule{
```

Comment:
Just to start the discussion (not really feedback for this PR)

The biggest drawback of the conversion funcs is, it's really easy to write bugs if you forget to map a field.

I wonder how we can test the conversion funcs accurately both ways? In the past I've seen reflection be used for this - reflect into both types, generate some value, assert it gets mapped. Are there other approaches you've seen?

Reply:
That kind of problem is supposed to be caught by either integration or request handler unit tests.
A specific test for these functions can be solved by having a generator for models - https://github.com/brianvoe/gofakeit. I brought that backend architecture meeting last year, and there was not objections to adopting this library (iirc, It is used in K6).